astropy/cosmology/funcs/comparison.py

```diff
@@ -0,0 +1,439 @@
+    auto-identified by ``Cosmology.from_format.registry``. Objects of these
+    kinds can still be checked for equality, but the correct format string must
+    be used.
+
+        >>> yml = cosmo2.to_format("yaml")
+        >>> cosmology_equal(cosmo1, yml, format=(None, "yaml"))
+        True
+
+    This also works with an array of 'format' matching the number of
+    cosmologies.
+
+        >>> cosmology_equal(mapping, yml, format=[True, "yaml"])
+        True
+    """
+    # Check parameter equality
+    if not allow_equivalent:
+        eq = (cosmo1 == cosmo2)
+
+    else:
+        # Check parameter equivalence
+        # The options are: 1) same class & parameters; 2) same class, different
+        # parameters; 3) different classes, equivalent parameters; 4) different
+        # classes, different parameters. (1) & (3) => True, (2) & (4) => False.
+        eq = cosmo1.__equiv__(cosmo2)
+        if eq is NotImplemented:
+            eq = cosmo2.__equiv__(cosmo1)  # that failed, try from 'other'
+
+        eq = False if eq is NotImplemented else eq
+
+    # TODO! include equality check of metadata
+
+    return eq
+
+
+@_comparison_decorator
+def _cosmology_not_equal(cosmo1: Any, cosmo2: Any, /, *, allow_equivalent: bool=False) -> bool:
+    r"""Return element-wise cosmology non-equality check.
+
+    .. note::
+
+        Cosmologies are currently scalar in their parameters.
+
+    Parameters
+    ----------
+    cosmo1, cosmo2 : |Cosmology|-like
+        The objects to compare. Must be convertible to |Cosmology|, as specified
+        by ``format``.
+
+    out : ndarray, None, optional
+        A location into which the result is stored. If provided, it must have a
+        shape that the inputs broadcast to. If not provided or None, a
+        freshly-allocated array is returned.
+
+    format : bool or None or str or tuple thereof, optional keyword-only
+        Whether to allow the arguments to be converted to a |Cosmology|. This
+        allows, e.g. a |Table| to be given instead a Cosmology. `False`
+        (default) will not allow conversion. `True` or `None` will, and will use
+        the auto-identification to try to infer the correct format. A `str` is
+        assumed to be the correct format to use when converting. ``format`` is
+        broadcast to match the shape of the cosmology arguments. Note that the
+        cosmology arguments are not broadcast against ``format``, so it cannot
+        determine the output shape.
+
+    allow_equivalent : bool, optional keyword-only
+        Whether to allow cosmologies to be equal even if not of the same class.
+        For example, an instance of :class:`astropy.cosmology.LambdaCDM` might
+        have :math:`\Omega_0=1` and :math:`\Omega_k=0` and therefore be flat,
+        like :class:`~astropy.cosmology.FlatLambdaCDM`.
+
+    See Also
+    --------
+    astropy.cosmology.cosmology_equal
+        Element-wise equality check, with argument conversion to Cosmology.
+
+    Examples
```

Comment:
It's not obvious that all those examples are needed if the function is private.

Reply:
True. A residual from when it was public. Removing entirely as ``cosmology_equal`` effectively covers all the examples.